lib/saluki-app/src/logging.rs

```diff
@@ -115,19 +133,47 @@ fn initialize_logging_inner(
             .replace(LoggingAPIHandler::new(shared_level_filter.clone(), reload_handle));
     }
 
+    let adp_log_file = std::env::var("DD_ADP_LOG_FILE").unwrap_or(DEFAULT_ADP_LOG_FILE.to_string());
+    let log_file_max_size = std::env::var("DD_LOG_FILE_MAX_SIZE")
+        .map(|s| s.parse::<u64>().unwrap_or(DEFAULT_LOG_FILE_MAX_SIZE))
+        .unwrap_or(DEFAULT_LOG_FILE_MAX_SIZE);
+    let log_file_max_rolls = std::env::var("DD_LOG_FILE_MAX_ROLLS")
+        .map(|s| s.parse::<usize>().unwrap_or(DEFAULT_LOG_FILE_MAX_ROLLS))
+        .unwrap_or(DEFAULT_LOG_FILE_MAX_ROLLS);
+    let file_appender = BasicRollingFileAppender::new(
+        adp_log_file,
+        RollingConditionBasic::new().max_size(log_file_max_size),
+        log_file_max_rolls,
+    )?;
+    let (file_nb, guard) = tracing_appender::non_blocking(file_appender);
+    let file_level_filter = EnvFilter::builder()
+        .with_default_directive(default_level.unwrap_or(LevelFilter::INFO).into())
+        .with_env_var("DD_LOG_LEVEL")
+        .from_env_lossy();
```

Comment:
This is going to cause us to log to a file by default, for anything using this... which we don't want to do.

We should -- in conjunction with my comment above about a single configuration type for logging stuff -- make this optional based on whether or not a log file is configured. `tracing_subscriber` supports optional layers through `impl<L> Layer for Option<L>` so we can just have something like `let maybe_file_layer = if enabled { Some(<actual layer here>) } else { None };` and then pop it in with `.with(maybe_file_layer)`.